frame/collective/src/lib.rs

```diff
@@ -349,7 +349,10 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
 		#[cfg(feature = "try-runtime")]
-		fn try_state(_n: BlockNumberFor<T>) -> Result<(), TryRuntimeError> {
+		fn try_state(
+			_n: BlockNumberFor<T>,
+			_s: frame_support::traits::TryStateSelect,
```

Comment:
This API is unfortunately not correct. 

You are right to pass in _something_ into the hook that helps it understand if it is fast or not, but `TryStateSelect` is not the right type here. 

`TryStateSelect` is meant to identify which pallets to execute, not how much time they should each consume. It is only interpreted by the `Executive` and should not be exposed to the end user here at all. 

What we instead want is a `TryStateSpeed` which you can for now assume to `bool` or `enum TryStateSpeed { Slow, Mid, Fast }`. 

Then, you are capable of selecting which pallets to run, and at what speed. 

As it stands now, I see this flaw as welll: 

A pallet could possibly see `RoundRobin(7)` as its try-state select. How should it interpret this? Answer: it cannot, because it is not the right audience for it.